code/modules/projectiles/pins.dm

```diff
@@ -222,4 +226,99 @@ Pins Below.
+	return !isStationLevel(T.z)
+
+var/list/wireless_firing_pins = list() //A list of all initialized wireless firing pins. Used in the firearm tracking console in guntracker.dm
+
+/obj/item/device/firing_pin/wireless
+	name = "wireless-control firing pin"
+	desc = "This firing pin is wirelessly controlled. On automatic mode it allow allows weapons to be fired on stun unless the alert level is elevated. Otherwise, it can be controlled from a firearm control console."
+	fail_message = "<span class='warning'>The wireless-control firing pin clicks!</span>"
+	var/registered_user = "Unregistered"
+	var/lockstatus = WIRELESS_PIN_AUTOMATIC
+
+/obj/item/device/firing_pin/wireless/examine_info(mob/user)
+	var/wireless_description
+	switch(lockstatus)
+		if(WIRELESS_PIN_AUTOMATIC)
+			wireless_description = "is in automatic mode"
+		if(WIRELESS_PIN_DISABLED)
+			wireless_description = "has locked the trigger"
+		if(WIRELESS_PIN_STUN)
+			wireless_description = "is in stun-only mode"
+		if(WIRELESS_PIN_LETHAL)
+			wireless_description = "is in unrestricted mode"
+	to_chat(user, "The wireless-control firing pin [wireless_description].")
+
+/obj/item/device/firing_pin/wireless/Initialize() //Adds wireless pins to the list of initialized wireless firing pins.
+	wireless_firing_pins += src
+	return ..()
+
+/obj/item/device/firing_pin/wireless/Destroy() //Removes the wireless pins from the list of initialized wireless firing pins.
+	wireless_firing_pins -= src
+	return ..()
+
+/*
+	The return of this pin_auth is dependent on the wireless pin's lockstatus. For the purposes of this, only energy projectiles
+	that have a taser_effect are considered "stun" projectiles. All ballistic firearms are considered as being lethal, regardless
+	of ammunition loaded. Behaviours should be as follows:
+	Automatic: The weapon will only fire on stun while the security level is Green or Blue. On Yellow, Red & Delta alert the weapon is unrestricted
+	Disabled: The weapon will not fire under any circumstance.
+	Stun-Only: The weapon will only fire on stun regardless of the current security level.
+	Lethal: The weapon will fire on all modes regardless of the current security level.
+*/
+/obj/item/device/firing_pin/wireless/pin_auth(mob/living/user)
+	if(lockstatus != WIRELESS_PIN_DISABLED)
+		if(lockstatus == WIRELESS_PIN_LETHAL)
+			return TRUE
+		if(istype(gun, /obj/item/gun/energy))
+			var/obj/item/gun/energy/thegun = gun
+			var/obj/item/projectile/energy/P = new thegun.projectile_type
+			if(!P)
+				return FALSE
+			if(P.taser_effect)
+				return TRUE
+			if(security_level != SEC_LEVEL_GREEN && security_level != SEC_LEVEL_BLUE && lockstatus == WIRELESS_PIN_STUN)
+				return FALSE
+		if(security_level != SEC_LEVEL_GREEN && security_level != SEC_LEVEL_BLUE && lockstatus != WIRELESS_PIN_STUN)
+			return TRUE
+	return FALSE
+
+/obj/item/device/firing_pin/wireless/proc/unlock(var/i) // Changes the current lockstatus of the weapon, and sends a message and sfx to whoever is holding it.
```

Comment:
this var/i should be a little more descriptive

Reply:
Fixed